### PR TITLE
Don't fall back to unvendored installs

### PR DESCRIPTION
**Pull request by vanschelven**
Falling back to unvendored installs is an anti-feature. In general this is the tool saying "I know you have explicitly requested to do X, but now that this has failed I will do the exact opposite of what you requested". In general, when a user explicitly requests X and this fails the correct course of action is to provide an error message.

Specifically:

* When vendored installs fail, unvendored installs will often also fail, but   may fail for different reasons. Example: build dependencies may not be available in the buildpack. This sends the user off on a wild goose chase (fixing the second problem, rather than the original one)

* When unvendored installs will in fact not fail (i.e. the intention of the now-removed code) the user's expectation that whatever is in the "vendor" directory is what's actually installed is violated. This violates reproducability of installs, and opens a can of worms security-wise (e.g. the vendored packages may have been scanned previously, whereas arbitrary downloads are generally not).

History: The anti-feature was introduced in https://github.com/cloudfoundry/python-buildpack/commit/518f933f15deb7bf8973796ea0dac54f36250b14, without providing
reasoning why this is desirable.

* [x] I have viewed signed and have submitted the Contributor License Agreement

* [x] I have made this pull request to the `develop` branch

* [ ] I have added an integration test

### CODE REVIEW

**Comment by vanschelven** [comment]
@brayanhenao @arjun024 any opinions on this?

**Comment by arjun024** [comment]
I agree with OP. Tried to find any pressing historical reasons but couldn't find anything.
Wondering if you all have any thoughts from your python experience @robdimsdale @thitch97

File: src/python/supply/supply.go [hunk 1]
```diff
@@ -703,16 +703,8 @@ func (s *Supplier) RunPipVendored() error {
...
-
-		if err := os.WriteFile(requirementsPath, originalReqs, 0644); err != nil {
-			return fmt.Errorf("could not overwrite modified requirements file: %v", err)
-		}
-
-		if err := s.RunPipUnvendored(); err != nil {
-			s.Log.Info("Running pip install failed. You need to include all dependencies in the vendor directory.")
-			return err
-		}
+        s.Log.Info("Running pip install failed. You need to include all dependencies in the vendor directory.")
```
Comment (arjun024) [inline src/python/supply/supply.go:706]: Could you indent these lines

Comment (robdimsdale) [reply src/python/supply/supply.go:706]: does `go fmt` work for this?

Comment (vanschelven) [reply src/python/supply/supply.go:706]: this is what happens when a Python programmer does some Go. Fixed.

**Review by robdimsdale — APPROVED** [review]

**Pull request merged**